bigquery_etl/glean_usage/templates/baseline_clients_first_seen_v1.init.sql

```diff
@@ -0,0 +1,39 @@
+{{ header }}
+{% from "macros.sql" import core_clients_first_seen %}
+
+CREATE TABLE IF NOT EXISTS
+  `{{ first_seen_table }}`
+PARTITION BY
+  first_seen_date
+CLUSTER BY
+  sample_id
+OPTIONS
+  (require_partition_filter = FALSE)
+AS
+WITH
+  baseline AS (
+    SELECT
+      client_info.client_id,
+      DATE(MIN(submission_timestamp)) as first_seen_date,
+      MIN(sample_id) as sample_id
+    FROM
+      `{{ baseline_table }}`
+    -- initialize by looking over all of history
+    WHERE
+      DATE(submission_timestamp) > "2010-01-01"
+    GROUP BY
+      client_id
+  )
+{% if fennec_id %}
+  ,
+  {{ core_clients_first_seen(migration_table) }}
+  SELECT
+    client_id,
+    COALESCE(core.first_seen_date, baseline.first_seen_date) as first_seen_date,
```

Comment:
It may be worth maintaining two additional dates here: `first_seen_baseline_date` and `first_seen_legacy_date` (NULL for apps without a legacy equivalent) would record the first observation by ping type. This would primarily be useful for debugging/validation.

Reply:
This deviates from the specification quite a bit. A `submission_date` field may work just as well, where the `submission_date` is the `first_seen_baseline_date` and the `first_seen_legacy_date` is effectively the `first_seen_date`. Debugging for legacy data would be all rows where the `submission_date <> first_seen_date`.

Reply:
> A submission_date field may work just as well, where the submission_date is the first_seen_baseline_date and the first_seen_legacy_date is effectively the first_seen_date.

That sounds fine. I don't want to lose momentum here by trying to reach perfection in naming. It may be better to avoid this issue altogether for now and keep just `first_seen_date` for simplicity. I don't feel strongly about it and will support any of the formulations that have been discussed here.

Reply:
One side-effect of adding another column is that it would increase the amount of data scanned every day, since there would be another date field (mostly the same in fenix, exactly the same for non-migration based tables).